### PR TITLE
zsteg: add package

### PR DESCRIPTION
**Pull request by D3vil0p3r**
It detects stegano-hidden data in PNG and BMP (unlike steghide that works only on JPEG files). So it complements steghide tool.

### CODE REVIEW

File: packages/ruby-iostruct/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# This file is part of BlackArch Linux ( https://www.blackarch.org/ ).
+# See COPYING for license details.
+
+_gemname=iostruct
```
Comment (noraj) [inline packages/ruby-iostruct/PKGBUILD:4]:
```suggestion
_gemname=${pkgname#ruby-}
```

and put after pkgname

File: packages/ruby-iostruct/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+pkgname=ruby-$_gemname
```
Comment (noraj) [inline packages/ruby-iostruct/PKGBUILD:5]:
```suggestion
pkgname=ruby-iostruct
```

File: packages/ruby-iostruct/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+pkgdesc='A Struct that can read/write itself from/to IO-like objects.'
+arch=('any')
+url='https://rubygems.org/gems/iostruct'
+license=('MIT')
+depends=('ruby')
+source=("https://rubygems.org/downloads/$_gemname-$pkgver.gem")
+noextract=("$_gemname-$pkgver.gem")
+sha512sums=('73dc0de2c1232b329eacef9f561b6f453b3a628c00ef5dee48becdd9a8fbd9df558fb61049e2af1117f4c27bd5325740567db2223016080856ebaecd8d32f4bd')
+
+package() {
```
Comment (noraj) [inline packages/ruby-iostruct/PKGBUILD:17]: Builds correctly.

File: packages/ruby-zpng/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# This file is part of BlackArch Linux ( https://www.blackarch.org/ ).
+# See COPYING for license details.
+
+_gemname=zpng
```
Comment (noraj) [inline packages/ruby-zpng/PKGBUILD:4]:
```suggestion
_gemname=${pkgname#ruby-}
```

and put after pkgname

File: packages/ruby-zpng/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+pkgname=ruby-$_gemname
```
Comment (noraj) [inline packages/ruby-zpng/PKGBUILD:5]:
```suggestion
pkgname=ruby-zpng
```

File: packages/ruby-zpng/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+pkgdesc='Pure ruby PNG file manipulation and validation.'
+arch=('any')
+url='https://github.com/zed-0xff/zpng'
+license=('MIT')
+depends=('ruby' 'ruby-rainbow')
+source=("https://rubygems.org/downloads/$_gemname-$pkgver.gem")
+noextract=("$_gemname-$pkgver.gem")
+sha512sums=('e8f4e5263217aca40b63352ac6337fc167b4af728c5a79b43623781c66f9ea61f62844a159b2ee5d087e5f7668d17ce0db661a1aa18866b7472fc16b073351e5')
+
+package() {
```
Comment (noraj) [inline packages/ruby-zpng/PKGBUILD:17]: builds correctly

File: packages/zsteg/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# This file is part of BlackArch Linux ( https://www.blackarch.org/ ).
+# See COPYING for license details.
+
+pkgname=zsteg
+_gemname=zsteg
```
Comment (noraj) [inline packages/zsteg/PKGBUILD:5]: _gemname is identical to pkgname so remove _gemname and replace it everywhere with pkgname

File: packages/zsteg/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+pkgver=v0.2.11.r0.g93cad73
+pkgrel=1
+pkgdesc='Detect stegano-hidden data in PNG and BMP.'
+arch=('any')
+groups=('blackarch' 'blackarch-stego' 'blackarch-anti-forensic')
+url='https://github.com/zed-0xff/zsteg'
+license=('MIT')
+depends=('ruby' 'ruby-zpng' 'ruby-iostruct')
```
Comment (noraj) [inline packages/zsteg/PKGBUILD:13]:
```suggestion
depends=('git' 'ruby' 'ruby-zpng' 'ruby-iostruct')
```

Comment (noraj) [reply packages/zsteg/PKGBUILD:13]: You are also not packaging ruby-prime: cf. https://github.com/zed-0xff/zsteg/blob/93cad73375a99ef06676243f6d70e0d931e1d8c7/Gemfile#L5

**Review by noraj — CHANGES_REQUESTED** [review]

File: packages/zsteg/PKGBUILD (hunk 8 above)
Comment (D3vil0p3r) [reply packages/zsteg/PKGBUILD:13]: @noraj Is it better if we insert `git` in `makedepends`?

File: packages/zsteg/PKGBUILD (hunk 7 above)
Comment (D3vil0p3r) [reply packages/zsteg/PKGBUILD]: Fixed

**Comment by noraj** [comment]
Thanks @D3vil0p3r for all your contributions.

**Comment by D3vil0p3r** [comment]
> Thanks @D3vil0p3r for all your contributions.

You are welcome

**Pull request merged**